src/main/webapp/app/exercises/shared/exercise/exercise.component.ts

```diff
@@ -82,10 +92,16 @@ export abstract class ExerciseComponent implements OnInit, OnDestroy {
 
     protected abstract loadExercises(): void;
 
+    protected abstract applyFilter(): void;
```

Comment:
Could we provide the implementation here as all implementations as similar?
Maybe something like:
```ts
applyFilter(exercises: [Exercise]): void {
    this.filteredExercises = exercises.filter((exercise) => this.filter.includeExercise(exercise));
    this.filteredExerciseCount.emit(this.filteredExercises.length);
}
```

Reply:
While this would be possible it would also mean that an exercise component (I'll use QuizExerciseComponent as an example) will have a `quizExercises` and a `filteredExercises` array. The first would be in QuizExerciseComponent and the latter in ExerciseComponent.
The only good solution I see would be to bring both to ExerciseComponent and use Generics for them. but this would be quite a large change in the ExerciseComponent classes, so I am not sure if I should actually do it.

Reply:
Hmmm, I see. Then let's leave it like this.